modules/provision/engine.go

```diff
@@ -15,9 +18,14 @@ type ReservationReadWriter interface {
 	Remove(id string) error
 }
 
+type Resulter interface {
```

Comment:
This name is terrible, could use another one

Reply:
Of course. change it to whatever feels appropriate.

Reply:
What about 
```suggestion
type  ResultCallback func(id string, result *Result) error
```

Reply:
Make this a type func would force me to pass a function to the constructor. 
I don't really like that idea.